azurerm/resource_arm_eventgrid_topic.go

```diff
@@ -0,0 +1,154 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/arm/eventgrid"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmEventGridTopic() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmEventGridTopicCreateUpdate,
+		Read:   resourceArmEventGridTopicRead,
+		Update: resourceArmEventGridTopicCreateUpdate,
+		Delete: resourceArmEventGridTopicDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"tags": tagsSchema(),
+
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"primary_access_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"secondary_access_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmEventGridTopicCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).eventGridTopicsClient
+
+	log.Printf("[INFO] preparing arguments for AzureRM EventGrid Topic creation.")
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	properties := eventgrid.Topic{
+		Location:        &location,
+		TopicProperties: &eventgrid.TopicProperties{},
+		Tags:            expandTags(tags),
+	}
+
+	_, createErr := client.CreateOrUpdate(resourceGroup, name, properties, make(chan struct{}))
+	err := <-createErr
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(resourceGroup, name)
```

Comment:
Is there any reason we cannot use `name` as ID and avoid making that extra API call? Otherwise this is susceptible to dataloss (and dangling resource) in case the second call fails for any reason.